coremltools/models/utils.py

```diff
@@ -1035,25 +1034,31 @@ def compile_model(model: _Union['_ct.models.MLModel', str, _Model_pb2.Model]) ->
     except:
         raise Exception("Unable to compile any Core ML models.")
 
-    # Check parameter
+    # Check model parameter
     if not isinstance(model, (str, _Model_pb2.Model, _ct.models.MLModel)):
```

Comment:
I understand the `if not isinstance(model, (str, _Model_pb2.Model, _ct.models.MLModel))` is kind of for "backward compatibility", but imho it would confuse new comers. How about
```
if isinstance(model, str):
    raise Exception
if isinstance(model, _ct.models.MLModel):
    raise Exception
if not isinstance(model, _Model_pb2.Model):
    raise Exception
```